### ISSUE TITLE
Update the `coreos/awscli` docker image according to the needs

### ISSUE THREAD

**Issue by mumoshu** [coreos/kube-aws#55]
## Initial ideas

It will settle between the below two images.

1. What is working in today's kube-aws:

   ```
   FROM alpine:3.1
   
   RUN apk --update add bash curl less groff jq python py-pip && pip install --upgrade awscli s3cmd && mkdir /root/.aws
   ```
   According to:
   * https://quay.io/repository/coreos/awscli/image/e33d9681c7bf50d591d9fcce3db7d21fbf261affde613e10c0abebb74f6b6a1c
   * https://quay.io/repository/coreos/awscli/build/49336018-390e-4c2a-aa44-1f84d67949ea

2. What is going to be needed for future work:

   cc @gianrubio
   
   e.g. https://github.com/coreos/kube-aws/issues/49#issue-188285087 which requires the `cfn-signal` command contained in [the aws-cfn-bootstrap package](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-helper-scripts-reference.html) to be included in a official-ish/controllable docker image(i.e. `quay.io/coreos/awscli`)
   
   I've talked about it with @pieterlange and this is what it would look like:

   ```
   FROM alpine
   RUN apk update && apk add python py-pip && rm -rf /var/cache/apk/*
   RUN pip install awscli
   RUN pip install https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz 
   ```

## Requirements

* Basically, the new image is ok as long as it:
  * keeps backward-compatibility(possibly including all the packages contained in the former image
  * contains only what is required for us as exactly as possible
  * sizes as small as possible
  * is not based on very out-dated alpine

## Implementation

It'll be something like:

```
FROM alpine:3.4

RUN apk --no-cache --update add bash curl less groff jq python py-pip && \
  pip install awscli s3cmd https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz && \
  apk del remove py-pip && \
  mkdir /root/.aws
```

**Comment by mumoshu** [comment]
`docker build` seems to fail because of outdated pip installed via `apt add`.
https://gist.github.com/mumoshu/5d7046950b6259eb68e1df5a406528e3

**Comment by mumoshu** [comment]
Added `pip install --upgrade pip` before `pip install`s, after `apk add py-pip` and it seems to work.
https://gist.github.com/mumoshu/300d21a82d26aa1cb5262e7c522f898b

**Comment by mumoshu** [comment]
It turns out that some cache files created by pip remains in `/root/.cache/pip`.

**Comment by mumoshu** [comment]
made it `pip install --no-cache-dir` so that the cache files disappear:

```
$ docker run -it --rm 99b5c774c7a7 bash
bash-4.3# find ~/
/root/
/root/.aws
/root/.cache
/root/.cache/pip
/root/.cache/pip/selfcheck.json
```

**Comment by mumoshu** [comment]
I had somehow misunderstood `apk add --no-cache` to not create `/var/cache/apt` at all.
However, It doesn't seem so:

```
bash-4.3# find /var
/var
/var/cache
/var/cache/apk
/var/cache/apk/APKINDEX.a2e6dac0.tar.gz
/var/cache/apk/APKINDEX.167438ca.tar.gz
/var/cache/misc
/var/tmp
/var/log
/var/empty
/var/lock
/var/lock/subsys
/var/spool
/var/spool/cron
/var/spool/cron/crontabs
/var/run
/var/local
/var/opt
/var/lib
/var/lib/udhcpd
/var/lib/apk
/var/lib/misc
```

**Comment by mumoshu** [comment]
We now have `quay.io/awscli:edge` for anyone who wants to test
https://quay.io/repository/coreos/awscli?tab=tags

Only less than 2%(0.4 MB) added to the original docker image!

**Comment by mumoshu** [comment]
I'll submit a pull request to add the dockerfile used for this shortly.

**Comment by pieterlange** [comment]
We should really keep `curl less groff jq ` in the image!

Can you use this as dockerfile instead?
```
FROM alpine
RUN apk update && apk add curl less groff jq python py-pip && rm -rf /var/cache/apk/*
RUN pip install awscli
RUN pip install https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz 
```

**Comment by mumoshu** [comment]
@pieterlange Thanks for looking into it!

See #56 for the actual Dockerfile used to build the image. In short, it does contain `curl less groff jq` 👍

**Comment by mumoshu** [comment]
FYI, we're going to have a dedicated repo for the docker image
https://github.com/coreos/awscli/pull/1

Please note that docker image tags won't change!

**Comment by mumoshu** [comment]
* According to the fact that automated builds from https://github.com/coreos/awscli to https://quay.io/repository/coreos/awscli is enabled
* and thanks to @pieterlange and @camilb's pull requests which fixed issues slipped into my initial implementation,

the `quay.io/coreos/awscli:latest` docker mageseems to be ready 🎉 

I'm closing this but feel free to reopen if it turned out that the initial scope of this issue isn't addressed yet.
